kolibri/auth/permissions.py

```diff
@@ -0,0 +1,349 @@
+"""
+This module defines the base classes for Kolibri's class-based Permissions system. Other apps can import the classes
+from this module in their own "permissions.py" module, and extend them, and then apply them to their own models.
+"""
+
+
+#####################################################################################################################
+### This first section consists of the base permissions classes: the ones likely to be imported and extended.     ###
+#####################################################################################################################
+
+class BasePermissions(object):
+    """
+    Base Permission class from which all other Permission classes should inherit.
+    By default, it denies permissions for everything. To be more permissive, override its methods.
+    Note that if you don't need to distinguish between create, update, and delete permissions, you can
+    simply override `_user_can_write_object` to define behavior for them in one place.
+    """
+
+    def user_can_create_object(self, user, obj):
+        """Returns True if this permission class grants <user> permission to create the provided <obj>.
+        Note that the object may not yet have been saved to the database (as this may be a pre-save check)."""
+        return self._user_can_write_object(user, obj)
+
+    def user_can_read_object(self, user, obj):
+        """Returns True if this permission class grants <user> permission to read the provided <obj>."""
+        return False
+
+    def user_can_update_object(self, user, obj):
+        """Returns True if this permission class grants <user> permission to update the provided <obj>."""
+        return self._user_can_write_object(user, obj)
+
+    def user_can_delete_object(self, user, obj):
+        """Returns True if this permission class grants <user> permission to delete the provided <obj>."""
+        return self._user_can_write_object(user, obj)
+
+    def _user_can_write_object(self, user, obj):
```

Comment:
Or [\_\_getattribute\_\_](https://docs.python.org/2/reference/datamodel.html#object.__getattribute__) instead of \_\_getattr\_\_